### PR TITLE
Revert "Revert "CPDNPQ480 ITT lead mentor registration""

### PR DESCRIPTION
**Pull request by shjohnson**
Reverts DFE-Digital/npq-registration#653

### CODE REVIEW

**Comment by Lockyy** [comment]
Happy to approve once specs are passing

**Comment by shjohnson** [comment]
Linked with this PR aswell https://github.com/DFE-Digital/early-careers-framework/pull/2936

**Review by Lockyy — APPROVED** [review]

File: app/lib/services/query_store.rb [hunk 1]
```diff
@@ -17,6 +17,10 @@ def itt_provider
     store["itt_provider"]
   end
 
+  def approved_itt_provider?
```
Comment (shjohnson) [inline app/lib/services/query_store.rb:20]: ahh this is what you meant by addition to the query store i guess

File: spec/features/sad_journeys/lead_mentor_journey_with_incorrect_course_spec.rb [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+      page.fill_in("Enter the name of the ITT provider you are working with", with: approved_itt_provider_legal_name)
+      page.click_button("Continue")
+    end
+
+    expect_page_to_have(path: "/registration/choose-your-npq", submit_form: true) do
+      expect(page).to have_text("Which NPQ do you want to do?")
+      page.choose("Senior leadership", visible: :all)
+    end
+
+    expect_page_to_have(path: "/registration/ineligible-for-funding", submit_form: false) do
```
Comment (shjohnson) [inline spec/features/sad_journeys/lead_mentor_journey_with_incorrect_course_spec.rb:66]: 🙌

**Review by peteryates — APPROVED** [review]
This looks good to me. Nothing in particular stands out (it didn't last time!)

**Pull request merged**